### PR TITLE
Add kernel cmdline parameter to enable or disable cgroups v2

### PR DESCRIPTION
**Pull request by devimc**
Set `agent.unified_cgroup_hierarchy` to `1` or `true` to enable cgroups v2 in the guest.
Set `agent.unified_cgroup_hierarchy` to `0` or `false` to disable cgroups v2 in the guest.
By default cgroups v2 is disabled.

cc @fidencio @crobinso

### CODE REVIEW

**Comment by fidencio** [comment]
cc @giuseppe as well.

**Comment by fidencio** [comment]
@devimc, I'll take some time Tomorrow to play with your change and give some feedback. Thanks for working on this.

**Comment by devimc** [comment]
@fidencio thanks, and don't forget to use the agent as init processes (`init=/usr/bin/kata-agent`) :smile:

**Comment by fidencio** [comment]
@devimc, firstly, let me start saying sorry for the long time it took me to actually test this MR and let me emphasize that I'll leave the code-review to someone more experienced with the code base, if you don't mind.

Now, regarding the tests. I've tested your patches against a F31 guest, which has cgroupsv2 support. I've tested both scenarios, forcing the usage cgroupsv1 and then forcing the usage of cgroupsv2, and I did **not** any issue. Your patches just work like a charm.

Nice work on this one! :-)

**Comment by devimc** [comment]
thanks @fidencio for taking the time and test this patch, I appreciate it :+1:

**Comment by devimc** [comment]
@amshinde @jodh-intel @lifupan  please take a look

File: agent.go [hunk 1]
```diff
@@ -1295,9 +1306,25 @@ func cgroupsMount() error {
...
+	if err != nil {
+		return err
+	}
+
+	var controllers string
+	for _, c := range strings.Fields(string(rawControllers)) {
+		controllers += fmt.Sprintf("+%v ", c)
+	}
+
+	return ioutil.WriteFile(cgroupSubtreeControlPath,
```
Comment (jodh-intel) [inline agent.go]: Nit: worth adding a link to the v2 doc for parity with the cgroup v1 URL doc link above?

**Comment by devimc** [comment]
@jodh-intel thanks, changes applied

**Pull request merged**